spec/lib/metasploit/framework/credential_collection_spec.rb

```diff
@@ -323,6 +339,253 @@
+            Metasploit::Framework::Credential.new(public: "test_public", private: nil),
+          )
+        end
+      end
+
+      context "when using password spraying" do
+        let(:password_spray) { true }
+
+        context "when :blank_passwords and :nil_password are true" do
+          let(:blank_passwords) { true }
+          let(:nil_passwords) { true }
+
+          context "with 2 additional_publics" do
+            let(:additional_publics) { [ "test_public1", "test_public2" ] }
+
+            # REMOVE BEFORE COMMIT: fails because no pwd spraying
+            specify  do
+              expect { |b| collection.each(&b) }.to yield_successive_args(
+                Metasploit::Framework::Credential.new(public: "test_public1", private: ""),
+                Metasploit::Framework::Credential.new(public: "test_public2", private: ""),
+                Metasploit::Framework::Credential.new(public: "test_public1", private: nil),
+                Metasploit::Framework::Credential.new(public: "test_public2", private: nil),
+              )
+            end
+          end
+        end
+
+        context "when given a user file" do
+          let(:user_file) do
+            filename = "user_file"
+            stub_file = StringIO.new("asdf\njkl\n")
+            allow(File).to receive(:open).with(filename, /^r/).and_return stub_file
+
+            filename
+          end
+
+          # REMOVE BEFORE COMMIT: this also yields the usernames as passwords for the additional_public
+          context "when given a password" do
+            let(:password) { "password" }
+
+            specify  do
+              expect { |b| collection.each(&b) }.to yield_successive_args(
+                Metasploit::Framework::Credential.new(public: "adsf", private: "password"),
+                Metasploit::Framework::Credential.new(public: "jkl", private: "password"),
+                Metasploit::Framework::Credential.new(public: "test_public", private: "password"),
+              )
+            end
+          end
+        end
+      end
+    end
+
+    context "when using password spraying" do
+      let(:password_spray) { true }
+      let(:username) { nil }
+      let(:password) { nil }
+
+      context "when :blank_passwords is true" do
+        let(:blank_passwords) { true }
+
+        context "with password (but no username)" do
+          let(:password) { "pass" }
+
+          # REMOVE BEFORE COMMIT: this yields empty creds (no username, no pass)
+          specify  do
+            expect { |b| collection.each(&b) }.to yield_successive_args()
+          end
+        end
+
+        # REMOVE BEFORE COMMIT: yields nothings, fails because of bug in method
+        context "with username (but no password)" do
+          let(:username) { "user" }
+
+          specify  do
+            expect { |b| collection.each(&b) }.to yield_successive_args(
+              Metasploit::Framework::Credential.new(public: username, private: ''),
```

Comment:
This fails because it generates no credential.
But in this case we have a username and are asking for a blank password so it should generate this. (this is the case I chose to describe in the issue)